### PR TITLE
WIP: GIST C2

### PR DESCRIPTION
**Pull request by brianedmonds90**

### CODE REVIEW

File: app/c2/gist.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        Retrieve all GIST posted results for a particular api key
+        :return:
+        """
+        results = await self._get_raw_gist_urls(comm_type='results')
+        result_content = await self._get_gist_content([result[0] for result in results])
+        await self._delete_gists([result[1] for result in results])
+        return result_content
+
+    async def get_beacons(self):
+        """ Retrieve all GIST beacons for a particular api key"""
```
Comment (khyberspache) [inline app/c2/gist.py]: Missing return value in doctoring

Comment (khyberspache) [reply app/c2/gist.py]: Also - this says "for a particular API key"... is the plan to eventually pass through the API key as an argument here?

File: app/c2/gist.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        """
+        Returns one of the GIST API keys to be encoded into the agent
+        :return: GIST api key
+        """
+        return self.key
+
+    async def get_results(self):
+        """
+        Retrieve all GIST posted results for a particular api key
+        :return:
```
Comment (khyberspache) [inline app/c2/gist.py]: This says "for a particular API key"... is the plan to eventually pass through the API key as an argument here?

File: app/c2/gist.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+class Gist(C2):
+
+    def __init__(self, module_info):
+        self.name = 'Gist'
+        self.key = module_info['config']['key']
+        self.c2_type = module_info['c2_type']
+
+    def encode_config_info(self):
+        """
+        Returns one of the GIST API keys to be encoded into the agent
```
Comment (khyberspache) [inline app/c2/gist.py]: I do not like this name.  Let's call it `get_key` or `get_cert` something

File: app/c2/gist.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    def _build_gist_content(comm_type, paw, files):
+        return {
+            "description": "{}-{}".format(comm_type, paw),
+            "public": False,
+            "files": files}
+
+    async def _get_gist_content(self, urls):
+        all_content = []
+        for url in urls:
+            headers = {'Authorization': 'token {}'.format(self.key)}
```
Comment (khyberspache) [inline app/c2/gist.py]: make this a function and replace all instances with the function call

File: app/c2/gist.py (hunk 2 above)
Comment (brianedmonds90) [reply app/c2/gist.py]: I don't think that's the plan. The docstring is misleading. I'll reword it.

**Comment by brianedmonds90** [comment]
Closing this for now as we are going to be white boarding our approach to alternative c2 next week

**Comment by brianedmonds90** [comment]
Old pr. There is a new one in place

**Pull request closed without merging**